src/assets.rs

```diff
@@ -195,23 +313,55 @@ impl<'a> AssetEditor<'a> {
     ///   * the given asset has the corresponding field.
     /// * Updates the mint information, if present in the given asset.
     /// * Sets as verified if either asset if verified.
-    pub fn update(&mut self, other: Asset) {
-        assert_eq!(self.asset.definition, other.definition);
-        if other.verified || !self.asset.verified {
+    pub fn update<L: Ledger>(mut self, other: Asset) -> Result<Self, KeystoreError<L>> {
+        let mut asset = self.store.load(&other.definition.code)?;
+        if other.verified || !asset.verified {
```

Comment:
Good question! We can indeed use the `asset` from the editor to update the existing asset in the `store`, but we also want to enable this function for standalone purposes such that we can pass any asset as `other` and use it to update the existing asset in the `store`.

Reply:
Makes sense, but I do wish there was a way to make it so editor flow always has an asset that is in storage and never has to load it.

Reply:
Actually shouldn't all asset updates be done by calling `assets.update(asset)` (calling the model itself) and calling the `editor.update()` should only be done with the same asset the editor was create with (previously asserted in the old code).

Reply:
Good point. Now I also think having an `other` parameter is redundant since the editor is always created with an asset. I'm going to remove `other` from `update` and `update_internal`.

Reply:
Oh, I almost forgot a case where having an `other` param is necessary. In `import_asset` in `lib.rs`:
```
session
    .assets
    .create_internal(
        asset.definition().clone(),
        asset.mint_info(),
        asset.verified(),
    )?
    .update_internal(asset)?
    .save()?;
```
An editor is created by `create_internal` with only the definition, mint info, and verified flag of `asset`, but with `update_internal(asset)`, we are able to update the editor asset with additional info (e.g., name), so I think it's still useful to have this parameter.

Reply:
Yeah this makes sense to me, I agree we need the parameter.

Reply:
> Yeah, agree `update` should be called only with the same asset definition the editor was originally created for, and should not do any additional `load`.

@jbearer Yeah, it makes sense to check that `other.definition` and `self.asset.definition` are the same and return an error if not. But I think we still need to call `load` in `update` to get the existing asset in the store, otherwise, we are not merging the given `other` asset with the asset in the store.

Reply:
`editor.asset` isn't necessarily loaded from the store. E.g., the `Assets::insert` function does the following:
1. Take an `asset` as input.
2. Use `asset` and `self.store` to construct an editor, so `editor.asset` isn't related to the store.
3. If an asset with the same code exists in the store, call `update` to merge `asset` with the store asset.

I think in step 3, checking if "an asset with the same code exists in the store" could be skipped, since we can do the check in `update` and get the store asset at the same time, saving a `load` call. But the `load` call in `update` is still necessary.

Reply:
The `AssetEditor` type represents the capability of incrementally updating a single asset. I really don't think it is the responsibility of `AssetEditor` to be loading new things from the database. That should be handled at a higher level, e.g. by the caller.

In your example, I think it is `insert`'s job to make sure the editor returned from `insert` already corresponds to the merging of the inserted asset with any potentially existing asset. This could look something like
```rust
let editor = if let Ok(store_asset) = store_asset {
  AssetEditor::new(&mut self.store, store_asset).update(asset)
} else {
  AssetEditor::new(&mut self.store, asset)
}
```